test/lint/lint-whitespace.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017-2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check for new lines in diff that introduce trailing whitespace or
+# tab characters instead of spaces.
+
+# We can't run this check unless we know the commit range for the PR.
+
+import argparse
+import os
+import re
+import sys
+
+from subprocess import check_output
+
+EXCLUDED_DIRS = ["depends/patches/",
+                 "contrib/guix/patches/",
+                 "src/leveldb/",
+                 "src/crc32c/",
+                 "src/secp256k1/",
+                 "src/minisketch/",
+                 "src/univalue/",
+                 "src/dashbls/",
+                 "src/immer/",
+                 "src/util/expected.h",
+                 "doc/release-notes/",
+                 "src/qt/locale"]
```

Comment:
24844: note/nit: "src/qt/locale" was added in 16988 but that's a "translations update" PR (i.e. DNM) so it's probably ok to simply fix it here.

Reply:
Also, `mapfile` shouldn't be a problem after #6429, the only shell scripts that'll remain then are `commit-script-check.sh`, `git-subtree-check.sh` and `lint-cppcheck-dash.sh` (former two remain shell scripts upstream[1] and the latter is a Dash-specific script)

[1] - [`commit-script-check.sh`](https://github.com/bitcoin/bitcoin/blob/ff873a20a7f48698da3334acc570d97a07016ddf/test/lint/commit-script-check.sh), [`git-subtree-check.sh`](https://github.com/bitcoin/bitcoin/blob/ff873a20a7f48698da3334acc570d97a07016ddf/test/lint/git-subtree-check.sh)